nominal/core/_utils/queueing.py

```diff
@@ -35,17 +35,19 @@ def put(self, item: _T_contra) -> None: ...
     def shutdown(self, immediate: bool = False) -> None: ...
 
 
-def _timed_batch(q: ReadQueue[BatchItem], max_batch_size: int, max_batch_duration: timedelta) -> Iterable[Batch]:
+def _timed_batch(
+    q: ReadQueue[BatchItem[_T]], max_batch_size: int, max_batch_duration: timedelta
+) -> Iterable[Batch[_T]]:
     """Yield batches of items from a queue, either when the batch size is reached or the batch window expires.
 
     Will not yield empty batches.
     """
-    batch: list[BatchItem] = []
+    batch: list[BatchItem[_T]] = []
     oldest_timestamp: IntegralNanosecondsUTC = MAX_INT64
     newest_timestamp: IntegralNanosecondsUTC = 0
-    next_batch_time = time.time() + max_batch_duration.total_seconds()
+    next_batch_time = time.monotonic() + max_batch_duration.total_seconds()
     while True:
-        now = time.time()
+        now = time.monotonic()
```

Comment:
you updated this to `time.monotonic()` but kept `time.time()` below when comparing. Want to do a quick sweep for `time.time`?